src/main/draft/tomcat.yaml

```diff
@@ -0,0 +1,457 @@
+              owner: root
+              group: root
+              mode: "000755"
+            /etc/tomcat/additional.properties:
+              content: |
+                cassandra.contactpoints={{cassandraContactPoints}}
+                jediscache.host={{jediscacheHost}}
+                jediscache.port=6379
+                org.jooq.no-logo=true
+              context:
+                cassandraContactPoints:
+                  Fn::ImportValue:
+                    !Sub "${cassandraStackName}-seed-nodes"
+                jediscacheHost:
+                  Fn::ImportValue:
+                    !Sub "${redisStackName}-redis-dns-name"
+              owner: root
+              group: root
+              mode: "000644"
+          commands:
+            patch_server_xml:
+              command: |
+                #!/bin/bash -xe
+                export SERVER_XML=/etc/tomcat/server.xml
+                sed -i "/<Valve className=\"org.apache.catalina.valves.AccessLogValve\"/c\    <\!-- <Valve className=\"org.apache.catalina.valves.AccessLogValve\" directory=\"logs\"" ${SERVER_XML}
+                sed -i "/<\/Host>/c\-->\n        <Valve className=\"ch.qos.logback.access.tomcat.LogbackValve\"/>\n      <\/Host>" ${SERVER_XML}
+            append_catalina_properties:
+              command: "cat /etc/tomcat/additional.properties >> /etc/tomcat/catalina.properties"
+        deploy_application:
+          commands:
+            download_war:
+              command: !Sub "aws s3 cp s3://${deployableBucketName}/${deployableFileKey} /opt/tomcat/webapps/${applicationContextPath}.war"
+        run_tomcat:
+          commands:
+             run_tomcat:
+               command: /opt/tomcat/bin/startup.sh
+
+  autoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      LaunchConfigurationName: !Ref launchConfiguration
+      MaxSize: 30
+      MinSize: 1
+      DesiredCapacity: 1
+      TargetGroupARNs:
+        - !Ref targetGroup
+      HealthCheckGracePeriod: 180
+      HealthCheckType: ELB
+      VPCZoneIdentifier:
+        - Fn::Select:
+          - 0
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 1
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 2
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} AS
+          PropagateAtLaunch: true
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT5M
+
+Outputs:
+  SecurityGroup:
+    Description: The redis security group, so that other security groups can allow ingress from themselves to it.
+    Value: !Ref securityGroup
+    Export:
+      Name: !Sub ${AWS::StackName}-security-group
```

Comment:
Perhaps this template should also export something like `BaseUrl` (just my first idea for the name, choose a better one if you have an idea) with a value that concatenates the `${publicLoadBalancerStackName}-public-lb-dns-name` with the `${applicationContextPath}`, so other stacks don't have to know the context path of this stack but can reference it if needed.

Reply:
I would YAGNI this, we do not have this use case anywhere.

Reply:
We've discussed this and the conclusion was, that an application that is intended to be called from other applications will
- most likely use the internal load balancer instead and thus require refactoring of this draft anyways
- can add the out and export parameters as needed while customizing.

It's the developers and reviewers job to remember to actually use out parameters instead of performing concatenation of "well known" load balancer and context path values on the call site.